src/helper_threading.py

```diff
@@ -23,20 +30,26 @@ def _thread_name_hack(self):
 
 
 class StoppableThread(object):
+    """A base class for threads giving them a custom stopping mechanism"""
     def initStop(self):
+        """Start the process of stopping the thread"""
```

Comment:
Perhaps I should have used a different name. This is just an initialiser, it does not stop anything. I avoided using `__init__` as this often used as a mixin, or maybe because of execution order, I don't remember the details. Perhaps I was overly cautious or simple there was something I haven't figured out how to do. If it can be done with a normal `__init__`, that would be preferable.